pcap.pyx

```diff
@@ -144,81 +147,88 @@ dltoff = { DLT_NULL:4, DLT_EN10MB:14, DLT_IEEE802:22, DLT_ARCNET:6,
         if pcap_ex_compile_nopcap(65535, dlt, &self.fcode, filter, 1, 0) < 0:
             raise IOError, 'bad filter'
+
     def filter(self, buf):
         """Return boolean match for buf against our filter."""
-        cdef char *p
-        cdef int n
-        if PyObject_AsCharBuffer(buf, &p, &n) < 0:
+        cdef u_char *p
+        cdef Py_ssize_t n
+        if PyObject_AsCharBuffer(buf, <const char**>&p, &n) < 0:
             raise TypeError
-        if bpf_filter(self.fcode.bf_insns, p, n, n) == 0:
-            return False
-        return True
+        return bpf_filter(self.fcode.bf_insns, p, <u_int>n, <u_int>n) != 0
+
     def __dealloc__(self):
         pcap_freecode(&self.fcode)
-            
+
+
 cdef class pcap:
     """pcap(name=None, snaplen=65535, promisc=True, timeout_ms=None, immediate=False)  -> packet capture object
-    
+
     Open a handle to a packet capture descriptor.
-    
+
     Keyword arguments:
     name      -- name of a network interface or dumpfile to open,
                  or None to open the first available up interface
     snaplen   -- maximum number of bytes to capture for each packet
     promisc   -- boolean to specify promiscuous mode sniffing
     timeout_ms -- requests for the next packet will return None if the timeout
                   (in milliseconds) is reached and no packets were received
-                  (Default: no timeout)  
+                  (Default: no timeout)
     immediate -- disable buffering, if possible
     """
     cdef pcap_t *__pcap
     cdef char *__name
     cdef char *__filter
     cdef char __ebuf[256]
     cdef int __dloff
-    
+
     def __init__(self, name=None, snaplen=65535, promisc=True,
                  timeout_ms=0, immediate=False):
         global dltoff
         cdef char *p
-        
+
         if not name:
             p = pcap_ex_lookupdev(self.__ebuf)
             if p == NULL:
                 raise OSError, self.__ebuf
         else:
-            p = name
-            
+            py_byte_name = name.encode('UTF-8')
+            p = py_byte_name
+
         self.__pcap = pcap_open_offline(p, self.__ebuf)
         if not self.__pcap:
             self.__pcap = pcap_open_live(pcap_ex_name(p), snaplen, promisc,
                                          timeout_ms, self.__ebuf)
         if not self.__pcap:
             raise OSError, self.__ebuf
-        
+
         self.__name = strdup(p)
         self.__filter = strdup("")
-        try: self.__dloff = dltoff[pcap_datalink(self.__pcap)]
-        except KeyError: pass
+        try:
+            self.__dloff = dltoff[pcap_datalink(self.__pcap)]
+        except KeyError:
+            pass
         if immediate and pcap_ex_immediate(self.__pcap) < 0:
             raise OSError, "couldn't enable immediate mode"
-    
+
     property name:
         """Network interface or dumpfile name."""
         def __get__(self):
-            return self.__name
+            return self.__name.decode('UTF-8')
```

Comment:
We should be careful to mention in the changelog that the return value of this function has changed. While in the past on python2 it would return a `str` now it's going to return a `unicode`.
I hope this does not break any existing code out there.